Python/chapter02/2.4 - Partition/miguel_2.4_soln.py

```diff
@@ -83,6 +83,18 @@ def get_node_at(self, index: int) -> Node:
                 return n
             n = n.next
 
+    def pop_head(self) -> Optional[Node]:
+        if self.head is None:
+            return self.head
+        h = self.head
+        self.head = self.head.next
+        return h
+
+    def append(self, ll: object):
+        if not isinstance(ll, LinkedList):
+            return TypeError
```

Comment:
shouldn't this be

```suggestion
            raise TypeError(...)
```

? the `return SomeError` pattern only occurs within `__eq__` and its comparison overload friends.